enclave/src/c_api/c_api.cc

```diff
@@ -1352,16 +1215,12 @@ XGB_DLL int XGBoosterPredict(BoosterHandle handle,
       (option_mask & 8) != 0,
       (option_mask & 16) != 0);
   preds = tmp_preds.HostVector();
-#ifdef __ENCLAVE__ // write results to host memory
-  //bst_float* result = (bst_float*) oe_host_malloc(preds.size()*sizeof(float));
-  //for (int i = 0; i < preds.size(); ++i) {
-  //    result[i] = preds[i];
-  //}
-  //*len = static_cast<xgboost::bst_ulong>(preds.size());
-  //*out_result = result;
-
   unsigned char key[CIPHER_KEY_SIZE];
-  EnclaveContext::getInstance().get_client_key((uint8_t*)key);
+  if (!EnclaveContext::getInstance().get_client_key((uint8_t*) key)) {
+    LOG(INFO) << "No client key found";
```

Comment:
See my other comment about this. Which level do we want to throw this error at? Might be easier if we do it in `get_client_key` so we only have to write that code once, instead of doing this check every time we call `get_client_key()`